### ISSUE TITLE
"Error: CUDA.jl does not yet support CUDA with ptxas 11.3.109" when using "JULIA_CUDA_USE_BINARYBUILDER=false"

### ISSUE THREAD

**Issue by ghost** [JuliaGPU/CUDA.jl#1089]
I am using CUDA V11.3.109 in a fresh install of manjaro linux. I set JULIA_CUDA_USE_BINARYBUILDER=false to see if the system version of CUDA is more stable on my laptop.

When I try to Pkg.test CUDA, I get this error:

```
julia> Pkg.test("CUDA")
     Testing CUDA
      Status `/tmp/jl_eoo81f/Project.toml`
  [79e6a3ab] Adapt v3.3.1
  [ab4f0b2a] BFloat16s v0.1.0
  [052768ef] CUDA v3.3.4
  [864edb3b] DataStructures v0.18.9
  [7a1cc6ca] FFTW v1.4.3
  [0c68f7d7] GPUArrays v7.0.1
  [a98d9a8b] Interpolations v0.13.2
  [872c559c] NNlib v0.7.27
  [a759f4b9] TimerOutputs v0.5.12
  [ade2ca70] Dates `@stdlib/Dates`
  [8ba89e20] Distributed `@stdlib/Distributed`
  [37e2e46d] LinearAlgebra `@stdlib/LinearAlgebra`
  [de0858da] Printf `@stdlib/Printf`
  [3fa0cd96] REPL `@stdlib/REPL`
  [9a3f8284] Random `@stdlib/Random`
  [2f01184e] SparseArrays `@stdlib/SparseArrays`
  [10745b16] Statistics `@stdlib/Statistics`
  [8dfed614] Test `@stdlib/Test`
      Status `/tmp/jl_eoo81f/Manifest.toml`
  [621f4979] AbstractFFTs v1.0.1
  [79e6a3ab] Adapt v3.3.1
  [13072b0f] AxisAlgorithms v1.0.0
  [ab4f0b2a] BFloat16s v0.1.0
  [fa961155] CEnum v0.4.1
  [052768ef] CUDA v3.3.4
  [d360d2e6] ChainRulesCore v1.3.0
  [34da2185] Compat v3.32.0
  [864edb3b] DataStructures v0.18.9
  [ffbed154] DocStringExtensions v0.8.5
  [e2ba6199] ExprTools v0.1.6
  [7a1cc6ca] FFTW v1.4.3
  [0c68f7d7] GPUArrays v7.0.1
  [61eb1bfa] GPUCompiler v0.12.8
  [a98d9a8b] Interpolations v0.13.2
  [692b3bcd] JLLWrappers v1.3.0
  [929cbde3] LLVM v4.2.0
  [2ab3a3ac] LogExpFunctions v0.2.5
  [872c559c] NNlib v0.7.27
  [6fe1bfb0] OffsetArrays v1.10.5
  [bac558e1] OrderedCollections v1.4.1
  [21216c6a] Preferences v1.2.2
  [74087812] Random123 v1.4.2
  [e6cf234a] RandomNumbers v1.5.3
  [c84ed2f1] Ratios v0.4.0
  [189a3867] Reexport v1.1.0
  [ae029012] Requires v1.1.3
  [276daf66] SpecialFunctions v1.6.0
  [90137ffa] StaticArrays v1.2.11
  [a759f4b9] TimerOutputs v0.5.12
  [efce3f68] WoodburyMatrices v0.5.3
  [f5851436] FFTW_jll v3.3.9+8
  [1d5cc7b8] IntelOpenMP_jll v2018.0.3+2
  [dad2f222] LLVMExtra_jll v0.0.7+0
  [856f044c] MKL_jll v2021.1.1+1
  [efe28fd5] OpenSpecFun_jll v0.5.5+0
  [0dad84c5] ArgTools `@stdlib/ArgTools`
  [56f22d72] Artifacts `@stdlib/Artifacts`
  [2a0f44e3] Base64 `@stdlib/Base64`
  [ade2ca70] Dates `@stdlib/Dates`
  [8bb1440f] DelimitedFiles `@stdlib/DelimitedFiles`
  [8ba89e20] Distributed `@stdlib/Distributed`
  [f43a241f] Downloads `@stdlib/Downloads`
  [b77e0a4c] InteractiveUtils `@stdlib/InteractiveUtils`
  [4af54fe1] LazyArtifacts `@stdlib/LazyArtifacts`
  [b27032c2] LibCURL `@stdlib/LibCURL`
  [76f85450] LibGit2 `@stdlib/LibGit2`
  [8f399da3] Libdl `@stdlib/Libdl`
  [37e2e46d] LinearAlgebra `@stdlib/LinearAlgebra`
  [56ddb016] Logging `@stdlib/Logging`
  [d6f4376e] Markdown `@stdlib/Markdown`
  [a63ad114] Mmap `@stdlib/Mmap`
  [ca575930] NetworkOptions `@stdlib/NetworkOptions`
  [44cfe95a] Pkg `@stdlib/Pkg`
  [de0858da] Printf `@stdlib/Printf`
  [3fa0cd96] REPL `@stdlib/REPL`
  [9a3f8284] Random `@stdlib/Random`
  [ea8e919c] SHA `@stdlib/SHA`
  [9e88b42a] Serialization `@stdlib/Serialization`
  [1a1011a3] SharedArrays `@stdlib/SharedArrays`
  [6462fe0b] Sockets `@stdlib/Sockets`
  [2f01184e] SparseArrays `@stdlib/SparseArrays`
  [10745b16] Statistics `@stdlib/Statistics`
  [fa267f1f] TOML `@stdlib/TOML`
  [a4e569a6] Tar `@stdlib/Tar`
  [8dfed614] Test `@stdlib/Test`
  [cf7118a7] UUIDs `@stdlib/UUIDs`
  [4ec0a83e] Unicode `@stdlib/Unicode`
  [e66e0078] CompilerSupportLibraries_jll `@stdlib/CompilerSupportLibraries_jll`
  [deac9b47] LibCURL_jll `@stdlib/LibCURL_jll`
  [29816b5a] LibSSH2_jll `@stdlib/LibSSH2_jll`
  [c8ffd9c3] MbedTLS_jll `@stdlib/MbedTLS_jll`
  [14a3606d] MozillaCACerts_jll `@stdlib/MozillaCACerts_jll`
  [83775a58] Zlib_jll `@stdlib/Zlib_jll`
  [8e850ede] nghttp2_jll `@stdlib/nghttp2_jll`
  [3f19e933] p7zip_jll `@stdlib/p7zip_jll`
Precompiling project...
  1 dependency successfully precompiled in 19 seconds (39 already precompiled)
     Testing Running tests...
┌ Error: CUDA.jl does not yet support CUDA with ptxas 11.3.109; please file an issue.
└ @ CUDA.Deps ~/.julia/packages/CUDA/lwSps/deps/discovery.jl:474
ERROR: LoadError: MethodError: Cannot `convert` an object of type Nothing to an object of type VersionNumber
Closest candidates are:
  convert(::Type{T}, ::T) where T at essentials.jl:205
Stacktrace:
  [1] CUDA.Deps.LocalToolkit(version::Any, dirs::Any)
    @ CUDA.Deps ~/.julia/packages/CUDA/lwSps/deps/bindeps.jl:52
  [2] find_local_cuda()
    @ CUDA.Deps ~/.julia/packages/CUDA/lwSps/deps/bindeps.jl:219
  [3] macro expansion
    @ ~/.julia/packages/CUDA/lwSps/deps/bindeps.jl:73 [inlined]
  [4] macro expansion
    @ lock.jl:209 [inlined]
  [5] toolkit()
    @ CUDA.Deps ~/.julia/packages/CUDA/lwSps/deps/bindeps.jl:19
  [6] toolkit_version()
    @ CUDA.Deps ~/.julia/packages/CUDA/lwSps/deps/bindeps.jl:241
  [7] toolkit_release()
    @ CUDA.Deps ~/.julia/packages/CUDA/lwSps/deps/bindeps.jl:248
  [8] top-level scope
    @ ~/.julia/packages/CUDA/lwSps/test/runtests.jl:114
  [9] include(fname::String)
    @ Base.MainInclude ./client.jl:444
 [10] top-level scope
    @ none:6
in expression starting at /home/ben/.julia/packages/CUDA/lwSps/test/runtests.jl:114
ERROR: Package CUDA errored during testing
Stacktrace:
  [1] pkgerror(msg::String)
    @ Pkg.Types /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/Types.jl:55
  [2] test(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; coverage::Bool, julia_args::Cmd, test_args::Cmd, test_fn::Nothing)
    @ Pkg.Operations /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/Operations.jl:1693
  [3] test(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; coverage::Bool, test_fn::Nothing, julia_args::Cmd, test_args::Cmd, kwargs::Base.Iterators.Pairs{Symbol, Base.TTY, Tuple{Symbol}, NamedTuple{(:io,), Tuple{Base.TTY}}})
    @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:343
  [4] test(pkgs::Vector{Pkg.Types.PackageSpec}; io::Base.TTY, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:80
  [5] test(pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:78
  [6] #test#63
    @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:76 [inlined]
  [7] test
    @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:76 [inlined]
  [8] #test#62
    @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:75 [inlined]
  [9] test(pkg::String)
    @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:75
 [10] top-level scope
    @ REPL[4]:1
```

**Comment by maleadt** [comment]
You did not mention which version of CUDA.jl you are using; probably an outdated one.
JULIA_CUDA_USE_BINARYBUILDER=false should generally not affect stability, so I advise against using that option.

**Comment by ghost** [comment]
Pkg.test outputs "3.3.4" if you trust it.

Pkg.test("CUDA") is crashing on my system, and I wanted to test with "JULIA_CUDA_USE_BINARYBUILDER=false" before filing a bug report. I'll take you advice and file the bug report without testing with a different version of CUDA.

I don't actually need to use my system version of CUDA, but in the output your team asked "please file an issue."

**Comment by maleadt** [comment]
> Pkg.test outputs "3.3.4" if you trust it.

Sorry, I somehow missed that.

> I don't actually need to use my system version of CUDA, but in the output your team asked "please file an issue."

Yes, thanks for filing the issue! I'm just trying to find out why people prefer a system CUDA installation, because in some point in the future we might remove the local discovery path.

Anyway, I found the issue reported here, and I'll fix it on the 3.3 branch.

**Comment by maleadt** [comment]
https://github.com/JuliaGPU/CUDA.jl/pull/1091/commits/7db98809687a9986fcc31d82d94c17a9d8ed2eca
